docs/versioned_docs/version-zenith/guides/snippets/github-google-cloud/python/main.py

```diff
@@ -0,0 +1,61 @@
+import dagger
+from dagger import dag, object_type, field, function
+
+@object_type
+class MyModule:
+
+    @function
+    def build(source: dagger.Directory) -> dagger.Container:
+        '''Build an image'''
+        return (
+            dag.container()
+            .from_("node:21")
+            .with_directory("/src", source)
+            .with_workdir("/src")
+            .with_exec(["cp", "-R", ".", "/home/node"])
+            .with_workdir("/home/node")
+            .with_exec(["npm", "install"])
+            .with_entrypoint(["npm", "start"])
+        )
+
+    @function
+    async def publish(self, source: dagger.Directory, registry: str, credential: dagger.Secret) -> str:
+        '''Publish an image'''
+        split = registry.split("/")
+        return await (
+            self.build(source)
+            .with_registry_auth(split[0], "_json_key", credential)
```

Comment:
This isn't in the current guide. Does the current guide work without it?

Reply:
Current guide relies on `docker login` auth on the host. This guide uses the JSON secret for client auth.